library/modules/Gui.cpp

```diff
@@ -130,529 +130,307 @@ static std::string getNameChunk(virtual_identity *id, int start, int end)
+        newFocusString = baseFocus;
+        newFocusString += "/Help";
+        focusStrings.push_back(newFocusString);
+    }
+    if (game->main_interface.settings.open) {
+        newFocusString = baseFocus;
+        newFocusString += "/Settings";
+        focusStrings.push_back(newFocusString);
+    }
+    if (game->main_interface.squad_equipment.open) {
+        newFocusString = baseFocus;
+        newFocusString += "/SquadEquipment";
+        focusStrings.push_back(newFocusString);
     }
-*/
-}
-
-/* TODO: understand how this changes for v50
-DEFINE_GET_FOCUS_STRING_HANDLER(dungeonmode)
-{
-    using df::global::adventure;
-
-    if (!adventure)
-        return;
-
-    focus += "/" + enum_item_key(adventure->menu);
-}
-
-DEFINE_GET_FOCUS_STRING_HANDLER(unitlist)
-{
-    focus += "/" + enum_item_key(screen->page);
-}
-
-DEFINE_GET_FOCUS_STRING_HANDLER(layer_military)
-{
-    auto list1 = getLayerList(screen, 0);
-    auto list2 = getLayerList(screen, 1);
-    auto list3 = getLayerList(screen, 2);
-    if (!list1 || !list2 || !list3) return;
-
-    focus += "/" + enum_item_key(screen->page);
-
-    int cur_list;
-    if (list1->active) cur_list = 0;
-    else if (list2->active) cur_list = 1;
-    else if (list3->active) cur_list = 2;
-    else return;
-
-    switch (screen->page)
-    {
-    case df::viewscreen_layer_militaryst::Positions:
-        {
-            static const char *lists[] = { "/Squads", "/Positions", "/Candidates" };
-            focus += lists[cur_list];
-            break;
-        }
-
-    case df::viewscreen_layer_militaryst::Equip:
-        {
-            focus += "/" + enum_item_key(screen->equip.mode);
-
-            switch (screen->equip.mode)
-            {
-            case df::viewscreen_layer_militaryst::T_equip::Customize:
-                {
-                    if (screen->equip.edit_mode < 0)
-                        focus += "/View";
-                    else
-                        focus += "/" + enum_item_key(screen->equip.edit_mode);
-                    break;
-                }
-            case df::viewscreen_layer_militaryst::T_equip::Uniform:
-                break;
-            case df::viewscreen_layer_militaryst::T_equip::Priority:
-                {
-                    if (screen->equip.prio_in_move >= 0)
-                        focus += "/Move";
-                    else
-                        focus += "/View";
-                    break;
-                }
-            }
-
-            static const char *lists[] = { "/Squads", "/Positions", "/Choices" };
-            focus += lists[cur_list];
-            break;
-        }
-
-    default:
-        break;
+    // squads should be last because it's the only one not exclusive with the others? or something?
+    if (game->main_interface.squads.open) {
+        newFocusString = baseFocus;
+        newFocusString += "/Squads";
+        focusStrings.push_back(newFocusString);
+    }
+
+    if (focusStrings.size() == 0) {
```

Comment:
since you don't clear focusStrings (and I'm not sure if you should), this can't truly be relied on. `if (!newFocusString.size()) {` would work though, I think